[GH-5951] Update Authentication > Guest Access (#6134)
mattermost/docs: source/configure/authentication-configuration-settings.rst

  :configjson: Enable
  :environment: N/A

  - **true**: Enables the guest account feature.
  - **false**: **(Default)** Disables the guest account feature.

Enable guest access
~~~~~~~~~~~~~~~~~~~

*Available in legacy Enterprise Edition E10 and E20*

+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+
| - **true**: Enables the guest account feature. See `Guest Accounts <https://docs.mattermost.com/onboard/guest-accounts.html>`__.                                                | - System Config path: **Authentication > Guest Access**             |

> | - **false**: **(Default)** Disables the guest account feature. See `Disable Guest Accounts <https://docs.mattermost.com/onboard/guest-accounts.html#disable-guest-accounts>`__. | - ``config.json`` setting: ``.GuestAccountsSettings.Enable: false`` |

Reviewer: Please see https://github.com/mattermost/docs/pull/6172#discussion_r1085382223 for guidance on the format for page link references in the docs.

*Available in legacy Enterprise Edition E10 and E20*

+---------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------+
| - **true**: Guests must use MFA. Guests without MFA are redirected to the setup page until MFA is configured. | - System Config path: **Authentication > Guest Access**                                       |
| - **false**: **(Default)** MFA is not required for guests.                                                    | - ``config.json`` setting: ``.GuestAccountsSettings.EnforceMultifactorAuthentication: false`` |
|                                                                                                               | - Environment variable: ``MM_GUESTACCOUNTSSETTINGS_ENFORCEMULTIFACTORAUTHENTICATION``         |
+---------------------------------------------------------------------------------------------------------------+-----------------------------------------------------------------------------------------------+

> | **Note**: This setting defaults to false and cannot be changed if MFA is not enforced for non-guest users.                                                                                                    |

Reviewer: ```suggestion
| **Note**: This setting defaults to **false** and cannot be changed if MFA isn't enforced for non-guest users.                                                                                                    |
```
